src/test/java/org/broadinstitute/barclay/argparser/CollectionArgumentUnitTests.java

```diff
@@ -0,0 +1,306 @@
+package org.broadinstitute.barclay.argparser;
+
+import org.testng.Assert;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.util.*;
+
+/**
+ * Tests for arguments that are collections (not to be confused with ArgumentCollection).
+ */
+public class CollectionArgumentUnitTests {
+
+    class UninitializedCollections {
+        @Argument
+        public List<String> LIST;
+        @Argument
+        public ArrayList<String> ARRAY_LIST;
+        @Argument
+        public HashSet<String> HASH_SET;
+        @PositionalArguments
+        public Collection<File> COLLECTION;
+
+    }
+
+    @DataProvider(name="uninitializedCollections")
+    public Object[][] uninitializedCollections() {
+        return new Object[][] {
+                // for these two tests, we expect the same results since both append and replace modes have
+                // the same behavior on collections with no initial values
+                {
+                    // replace mode
+                    Collections.EMPTY_SET,
+                    new String[] {"--LIST","L1", "--LIST","L2", "--ARRAY_LIST","S1", "--HASH_SET","HS1", "P1", "P2"},
```

Comment:
I would extract the redundant declaration of the `String[]` as a variable in this dataprovider so it's clear that it's being reused between the tests.